src/Features/CSharp/Portable/UseExpressionBody/UseExpressionBodyCodeRefactoringProvider.cs

```diff
@@ -248,7 +248,11 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
                 CancellationToken cancellationToken)
             {
                 // Process all declaration nodes in reverse to handle nested declaration updates properly.
-                declarationsToFix = declarationsToFix.Reverse();
+                // NOTE: We need to realize the declarations with 'ToArray' call here
+                // to ensure we strongly hold onto the nodes so that 'TrackNodes'
+                // invoked below, which does tracking based off a ConditionalWeakTable,
+                // tracks the nodes for the entire duration of this method.
+                declarationsToFix = declarationsToFix.Reverse().ToArray();
```

Comment:
consider having 'declarationsToFix' be an array/immutablearray itself.

Reply:
Done